packages/ai/core/tool/tool.ts

```diff
@@ -56,21 +63,16 @@ If not provided, the tool will not be executed automatically.
 @args is the input of the tool call.
 @options.abortSignal is a signal that can be used to abort the tool call.
    */
-  execute?: (
+  execute: (
```

Comment:
Removing the optionality here will break important functionality, namely tools without execute.

Reply:
I got two alternatives idea for this issue

- Create an isolated tool in different name (eg: `etool` specifically for `generateCode`

- add one more option to `toolChoice: "none" | "auto" | "required" | "code"` (this `code` helps to make `execute` strictly required

Give me a feedback, I'll rework on this